fence/agents/azure_arm/fence_azure_arm.py

```diff
@@ -0,0 +1,136 @@
+#!@PYTHON@ -tt
+
+import sys, re, pexpect
+import logging
+import atexit
+sys.path.append("/usr/share/fence")
+from fencing import *
+from fencing import fail, fail_usage, EC_TIMED_OUT, run_delay
+
+#BEGIN_VERSION_GENERATION
+RELEASE_VERSION="4.0.24.6-7e576"
+BUILD_DATE="(built Thu Nov 3 15:43:03 STD 2016)"
+REDHAT_COPYRIGHT="Copyright (C) Red Hat, Inc. 2004-2010 All rights reserved."
+#END_VERSION_GENERATION
+
+compute_client = None
+
+def get_nodes_list(conn, options):
+    result = {}
+    if compute_client:
+        rgName = options["--resourceGroup"]
+        vms = compute_client.virtual_machines.list(rgName)
+        for vm in vms:
+            result[vm.name] = ("", None)
+
+    return result
+
+def get_power_status(conn, options):
+    logging.info("getting power status for VM " + options["--plug"])
+
+    if compute_client:
+        rgName = options["--resourceGroup"]
+        vmName = options["--plug"]
+
+        powerState = "unknown"
+        vmStatus = compute_client.virtual_machines.get(rgName, vmName, "instanceView")
+        for status in vmStatus.instance_view.statuses:
+            if status.code.startswith("PowerState"):
+                powerState = status.code
+                break
+
+        logging.info("Found power state of VM: " + powerState)
+        if powerState == "PowerState/running":
+            return "on"
+
+    return "off"
+
+def set_power_status(conn, options):
+    logging.info("setting power status for VM " + options["--plug"] + " to " + options["--action"])
+
+    if compute_client:
+        rgName = options["--resourceGroup"]
+        vmName = options["--plug"]
+
+        if (options["--action"]=="off"):
+            logging.info("Deallocating " + vmName + "in resource group " + rgName)
+            compute_client.virtual_machines.deallocate(rgName, vmName)
+        elif (options["--action"]=="reboot"):
```

Comment:
You will not enter 'reboot' part because reboot is translated to off, check if OFF, on, check if ON. If you want to have a reboot function that cannot be verified (like you have) you have to create a new function, pass it to fence_action(..) and add option --method (look at fence_ipmi)